### PR TITLE
Update Salmon to 0.4.0

### PR DESCRIPTION
**Pull request by Blahah**
this is a WIP - I will add bottles

### CODE REVIEW

File: salmon.rb [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
 
-  url "https://github.com/kingsfordgroup/sailfish/archive/v0.3.0.tar.gz"
-  sha1 "69f5752d79b3d8c8ba477ac8ba8d96350a68ebc4"
-  head "https://github.com/kingsfordgroup/sailfish.git", :branch => "develop"
-
-  bottle do
-    root_url "https://downloads.sf.net/project/machomebrew/Bottles/science"
-    sha1 "52e52d191915334cc33f7613fdcce3e0b4d71d1f" => :yosemite
-    sha1 "a04b4b5eadde9c0e93c1fc6253ea8af6f3d1cb6a" => :mavericks
-  end
```
Comment (sjackman) [inline salmon.rb]: No need to delete the bottle section. The BrewTestBot CI automatically updates the bottles.

File: salmon.rb [hunk 2]
```diff
@@ -1,16 +1,10 @@
 class Salmon < Formula
-  homepage "http://www.cs.cmu.edu/~ckingsf/software/sailfish/downloads.html"
+  homepage "http://salmon.readthedocs.org/en/latest/"
```
Comment (sjackman) [inline salmon.rb]: I prefer the GitHub homepage: https://github.com/COMBINE-lab/salmon
It links to readthedocs as well

File: salmon.rb [hunk 3]
```diff
@@ -1,16 +1,10 @@
...
-  head "https://github.com/kingsfordgroup/sailfish.git", :branch => "develop"
-
-  bottle do
-    root_url "https://downloads.sf.net/project/machomebrew/Bottles/science"
-    sha1 "52e52d191915334cc33f7613fdcce3e0b4d71d1f" => :yosemite
-    sha1 "a04b4b5eadde9c0e93c1fc6253ea8af6f3d1cb6a" => :mavericks
-  end
+  url "https://github.com/COMBINE-lab/salmon/archive/v0.4.0.tar.gz"
+  sha1 "3552ab0e8bb85be56b12f1b34356daaf48a43469"
+  head "https://github.com/COMBINE-lab/salmon.git", :branch => "master"
```
Comment (sjackman) [inline salmon.rb]: `master` is the default. You can delete `, :branch => "master"`

File: salmon.rb [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
-  sha1 "69f5752d79b3d8c8ba477ac8ba8d96350a68ebc4"
-  head "https://github.com/kingsfordgroup/sailfish.git", :branch => "develop"
-
-  bottle do
-    root_url "https://downloads.sf.net/project/machomebrew/Bottles/science"
-    sha1 "52e52d191915334cc33f7613fdcce3e0b4d71d1f" => :yosemite
-    sha1 "a04b4b5eadde9c0e93c1fc6253ea8af6f3d1cb6a" => :mavericks
-  end
+  url "https://github.com/COMBINE-lab/salmon/archive/v0.4.0.tar.gz"
+  sha1 "3552ab0e8bb85be56b12f1b34356daaf48a43469"
```
Comment (sjackman) [inline salmon.rb]: Please use `sha256` instead.

File: salmon.rb [hunk 5]
```diff
@@ -1,16 +1,10 @@
 class Salmon < Formula
```
Comment (sjackman) [inline salmon.rb]: Please add `desc "Transcript-level quantification from RNA-seq reads"`

**Comment by sjackman** [comment]
You may need to add `ENV.libstdcxx` to get the build to succeed on Mountain Lion.

File: salmon.rb (hunk 1 above)
Comment (Blahah) [inline salmon.rb]: very cool!

**Comment by sjackman** [comment]
http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/2136/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit_Homebrew_science_salmon/
```
Error: 1 problems in 1 formulae
homebrew/science/salmon:
 * 27: Trailing whitespace was found
```
Use `brew audit --strict salmon`

**Comment by sjackman** [comment]
Please squash commits. See https://github.com/Homebrew/homebrew-science/blob/master/CONTRIBUTING.md

File: salmon.rb [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
-  homepage "http://www.cs.cmu.edu/~ckingsf/software/sailfish/downloads.html"
+  homepage "https://github.com/COMBINE-lab/salmon"
+  desc "Transcript-level quantification from RNA-seq reads"
```
Comment (sjackman) [inline salmon.rb]: The `desc` should go above the `homepage`. I know this is a niggly detail. Insert quote
> Foolish consistency is the hobgoblin of little minds

File: salmon.rb [hunk 7]
```diff
@@ -24,7 +23,8 @@ class Salmon < Formula
   def install
     # Fix error: Unable to find the requested Boost libraries.
     ENV.deparallelize
-
+    ENV.libstdcxx
```
Comment (sjackman) [inline salmon.rb]: If this line is indeed necessary, it should include the build error that it fixes in a comment. The only real way to check that is by submitting the formula without this line, and seeing if it fails.

**Pull request closed without merging**